src/movement.rs

```diff
@@ -50,45 +66,102 @@ pub fn player_controller(
         With<Player>,
     >,
     time: Res<Time>,
+    game_meta: Res<GameMeta>,
+    windows: Res<Windows>,
+    left_movement_boundary: Res<LeftMovementBoundary>,
 ) {
-    for (mut state, stats, mut transform, facing_option, input) in query.iter_mut() {
-        if *state != State::Idle && *state != State::Running {
-            break;
-        }
+    let players_x = query
+        .iter()
+        .map(|(_, _, transform, _, _)| transform.translation.x)
+        .collect::<Vec<_>>();
 
-        let mut dir = Vec2::ZERO;
+    // Compute the new direction vectors; can be None if the state is not (idle or running).
+    //
+    let mut player_dirs = query
+        .iter()
+        .map(|(state, stats, transform, _, input)| {
+            if *state != State::Idle && *state != State::Running {
+                None
+            } else {
+                let mut dir = Vec2::ZERO;
 
-        if input.pressed(PlayerAction::Move) {
-            dir = input.action_axis_pair(PlayerAction::Move).unwrap().xy();
-        }
+                if input.pressed(PlayerAction::Move) {
+                    dir = input.action_axis_pair(PlayerAction::Move).unwrap().xy();
+                }
 
-        // Apply speed
-        dir = dir * stats.movement_speed * time.delta_seconds();
+                // Apply speed
+                dir = dir * stats.movement_speed * time.delta_seconds();
 
-        //Restrict player to the ground
-        let new_y = transform.translation.y + dir.y + consts::GROUND_OFFSET;
+                let new_x = transform.translation.x + dir.x;
 
-        if new_y >= consts::MAX_Y || new_y <= consts::MIN_Y {
-            dir.y = 0.;
-        }
+                // The dir.x condition allows some flexibility (e.g. in case of knockback), given
+                // the current state of development. To be removed once the movement logic is
+                // stabilized.
+                //
+                if dir.x < 0. && new_x < left_movement_boundary.0 {
+                    dir.x = 0.;
+                }
+                println!("{}", new_x);
```

Comment:
Remember to remove in future

Reply:
I've opened an issue (#63), which describes the general note, and includes a reference to this condition :smile:.